STIPS/STIP-006.md

```diff
@@ -0,0 +1,296 @@
+- [ ] Should we name this TradeModuleV2 or TradeModuleWithRebates (GeneralIndexModuleV2 or GeneralIndexModuleWithRebates)?
+- [ ] How do we prevent managers charging an 100% fee and rugging the Set on a trade?
+    - We will only allow governance to change rebate %s
+
+## Feasibility Analysis
+
+#### Option 1
+Adding a managerFee % which is assessed on top of the protocol fee %. This will be least code changes to the TradeModule and General Index Module. To avoid rugging, there will be a maxManagerFee enforced.
+```
+uint256 protocolFeeTotal = getModuleFee(TRADE_MODULE_PROTOCOL_FEE_INDEX, _exchangedQuantity);
+
+// Get manager fee here
+```
+
+#### Option 2
+Governance specifies a protocol fee % and a rebate split %. The split gives the manager a rebate on a percentage of protocol fees. Manager has no say and cannot rug.
+
+```
+uint256 protocolFeeSplit = controller.getModuleFee(address(this), PROTOCOL_FEE_SPLIT_INDEX);
+
+uint256 protocolDirectFee = controller.getModuleFee(address(this), PROTOCOL_DIRECT_FEE_SPLIT_INDEX);
+protocolFee = protocolDirectFee.preciseMul(protocolFeeSplit);
+rebateFee = protocolDirectFee.sub(protocolFee);
+```
+
+#### Recommendation
+Option 4 gives us protection against rugging while providing us a base rebate mechanism. This means we will charge a protocol fee to start in order to activate rebates. E.g. start with a 10 bps protocol fee and 5 bps rebate. 
+
+Because the rebate % is fixed by governance across all Sets, the only variable is AUM of the Set. The more user AUM vs manager AUM the more rebates managers receive (which eventually will become negative).
+
+#### Future work
+Tiered rebates: With this base rebate mechanism, manager contracts can be built on top in the future. Individual managers will always have the ability to specify a fee recipient address as their own, but manager contracts for a specific product (e.g. social trading) can abstract away the fee recipient to a shared peripheral contract. This central trading contract tracks cumulative volume done through trading all the Sets that are linked, and perform calculations that split the rebates collected. This way, at the base module level, rebates are flat across Sets but at the manager contract level, rebates can be tiered
+
+Negotiated rebates: Updating the TradeModule to a V3 to allow each Set to have its own rebate percentage (instead of a global one for now). This could be negotiated with large managers/applications that use the protocol.
+
+## Timeline
+- Spec + review: 2 days
+- Implementation: 2 days
+- Internal review: 1 days
+- Deployment scripts: 1 day
+- Deploy to testnet: 1 day
+- Testing: 1 day
+- Write docs: 1 day
+
+## Checkpoint 1
+
+**Reviewer**:
+
+## Proposed Architecture Changes
+
+### TradeModuleV2
+- Inherit TradeModule
+- Override `trade` to update ComponentExchanged event
+- Override `_accrueProtocolFee`
+- Override initialize to add `managerRebateRecipient`
```

Comment:
I'm not sure it'll be this simple since this will require a different interface so isn't a clean override, more of an overload. Also we'll have to track this state somewhere.